### PR TITLE
Refine image modal for signing component

### PR DESCRIPTION
**Pull request by DabeerAzeez**
- Organize the different buttons within the modal ("add" and "clear" buttons are held within the modal body, a singular "save" button is in the modal footer to the right)
- Add a 'no images selected' graphic
- Set up image previews within the modal

### CODE REVIEW

File: src/app/pages/signing/image-modal/image-modal.component.ts [hunk 1]
```diff
@@ -12,15 +8,16 @@ import { ErrorAlertComponent } from '../error-alert/error-alert.component';
   styleUrls: ['./image-modal.component.css'],
 })
 export class ImageModalComponent {
-  modalRef: BsModalRef;
+  static maxFileSize = 8 * 1000000; // 8 MB in bytes
```
Comment (ricetech) [inline src/app/pages/signing/image-modal/image-modal.component.ts]: I'm using `8 * 1024 * 1024` in `storage.rules`. We should make these match
```suggestion
  static maxFileSize = 8 * 1024 * 1024; // 8 MiB in bytes
```

File: src/app/pages/signing/image-modal/image-modal.component.ts [hunk 2]
```diff
@@ -30,21 +27,26 @@ export class ImageModalComponent {
     this.imageModal.hide();
   }
 
-  fileNames(fileInput: any) {
-    this.selectedImagesNames = [];
-    this.selectedImages = [];
+  onSelectFiles(e) {
+    const files = e.target.files;
```
Comment (ricetech) [inline src/app/pages/signing/image-modal/image-modal.component.ts]: Can you check to see if this triggers a warning/error in code inspection? I can't check rn

File: src/app/pages/signing/image-modal/image-modal.component.ts [hunk 3]
```diff
@@ -30,21 +27,26 @@ export class ImageModalComponent {
...
+    if (files) {
+      for (const item of files) {
+        if (item.size > ImageModalComponent.maxFileSize) {
+          this.ErrorAlertComponent.addImageSizeLimit(item.name);
+          return;
+        }
+
+        const reader = new FileReader();
+        reader.readAsDataURL(item);
+        reader.onload = (event: any) => {
```
Comment (ricetech) [inline src/app/pages/signing/image-modal/image-modal.component.ts]: Can we avoid using `any` lol

**Review by ricetech — CHANGES_REQUESTED** [review]

File: src/app/pages/signing/image-modal/image-modal.component.ts (hunk 2 above)
Comment (DabeerAzeez) [reply src/app/pages/signing/image-modal/image-modal.component.ts]: Just inspected `image-modal.component.ts`. No suspicious code found.

File: src/app/pages/signing/image-modal/image-modal.component.ts (hunk 3 above)
Comment (DabeerAzeez) [reply src/app/pages/signing/image-modal/image-modal.component.ts]: I think usually there isn't a type declaration ([here](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/onload#:~:text=%20reader.onload%20%3D%20function(e)%20%7B%20) is the `FileReader.onload` documentation). I just removed it and Webstorm shows this:

image

Thoughts?

Comment (ricetech) [reply src/app/pages/signing/image-modal/image-modal.component.ts]: That's because you're reading code written in JavaScript, not TypeScript.

Comment (DabeerAzeez) [reply src/app/pages/signing/image-modal/image-modal.component.ts]: I added the type declaration that WebStorm suggested

**Review by kishanmanoharan — APPROVED** [review]

**Review by ricetech — APPROVED** [review]

**Pull request merged**